pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdBookies.java

```diff
@@ -35,7 +38,8 @@ private class GetAll extends CliCommand {
 
         @Override
         void run() throws Exception {
-            print(getAdmin().bookies().getBookiesRackInfo());
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            print(gson.toJson(getAdmin().bookies().getBookiesRackInfo()));
```

Comment:
I found this call is already back by:

```java
    <T> void print(T item) {
        try {
            if (item instanceof String) {
                System.out.println(item);
            } else {
                System.out.println(writer.writeValueAsString(item));
            }
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
```

... where

```java
private static ObjectMapper mapper = ObjectMapperFactory.create();
```

Isn't the current code pretty print? Otherwise, we may try to fix the Jackson ObjectMapper config instead of wrapping up one more layer on Gson to hack.

Reply:
Hi @tisonkun, you are right but since BookiesRackConfiguration class extends TreeMap, it goes to the first definition of print 

`<K, V> void print(Map<K, V> items) {
        for (Map.Entry<K, V> entry : items.entrySet()) {
            print(entry.getKey() + "    " + entry.getValue());
        }
    }`

where it is serialized to string already. So it does not do pretty print.